`MapResult` object is multiprocessing library is of wrong typing.

> moved from https://github.com/python/mypy/issues/6883

```python
import multiprocessing
import time


def worker(i):
    time.sleep(1)
    return f"hello {i}"


with multiprocessing.Pool(processes=8) as pool:
    result_objs = pool.map_async(worker, list(range(10)))
    while not result_objs.ready():
        print(f"num left: {result_objs._number_left}")
        time.sleep(0.2)
```

The code works fine, and the type if `map_async` return is `<class 'multiprocessing.pool.MapResult'>`, but mypy report this error message.

```shell
test_pool.py:25: error: "ApplyResult[List[Any]]" has no attribute "_number_left"
```

--- a/stdlib/3/multiprocessing/pool.pyi
+++ b/stdlib/3/multiprocessing/pool.pyi
@@ -17,6 +17,8 @@ class ApplyResult(Generic[_T]):
 # alias created during issue #17805
 AsyncResult = ApplyResult
 
+class MapResult(ApplyResult[List[_T]]): ...
+
 _IMIT = TypeVar('_IMIT', bound=IMapIterator)
 
 class IMapIterator(Iterator[_T]):
@@ -50,7 +52,7 @@ class Pool(ContextManager[Pool]):
                   iterable: Iterable[_S] = ...,
                   chunksize: Optional[int] = ...,
                   callback: Optional[Callable[[_T], None]] = ...,
-                  error_callback: Optional[Callable[[BaseException], None]] = ...) -> AsyncResult[List[_T]]: ...
+                  error_callback: Optional[Callable[[BaseException], None]] = ...) -> MapResult[List[_T]]: ...
     def imap(self,
              func: Callable[[_S], _T],
              iterable: Iterable[_S] = ...,

